core/java/iesi-core/src/main/java/io/metadew/iesi/script/action/http/HttpExecuteRequest.java

```diff
@@ -136,10 +135,10 @@ protected String getKeyword() {
         } else if (dataType instanceof Text) {
             return Arrays.stream(dataType.toString().split(","))
                     .map(this::buildHttpHeader).collect(Collectors.toList());
-        } else if (dataType instanceof InMemoryDatasetImplementation) {
-            return InMemoryDatasetImplementationService
+        } else if (dataType instanceof DatasetImplementation) {
+            return DatasetImplementationHandler
                     .getInstance()
-                    .getDataItems((InMemoryDatasetImplementation) dataType, getExecutionControl()
+                    .getDataItems((DatabaseDatasetImplementation) dataType, getExecutionControl()
```

Comment:
Why cast to DatabaseDatasetImplementation? there should be no cast

Reply:
getDataItems have parameter DatasetImplementation, so we should cast to DatasetImplementation instead of DatabaseDatasetImplementation.